src/vs/workbench/contrib/notebook/browser/view/cellParts/cellComments.ts

```diff
@@ -108,7 +107,7 @@ export class CellComments extends CellContentPart {
 		if (this._commentThreadWidget.value) {
 			if (!info) {
 				this._commentThreadDisposables.clear();
-				this._commentThreadWidget.dispose();
+				this._commentThreadWidget.value = undefined;
```

Comment:
I think now `this._commentThreadWidget` is never disposed. It can be added to the object's disposables by doing this when it's created:

```ts
	private readonly _commentThreadWidget = this._register(new MutableDisposable<CommentThreadWidget<ICellRange>>);
```

Reply:
Good point. I added that to the constructor, as that makes it a bit easier to follow in which order things happen, and I see other _register() calls in thee constructor. Let me know what you think.